test/verify/machineslib.py

```diff
@@ -178,13 +178,41 @@ def prepareStorageDeviceOnISCSI(m, target_iqn, orig_iqn=None):
 class TestMachines(NetworkCase):
     created_pool = False
     provider = None
+    tmp_storage = "/mnt/libvirt"
 
     def setUp(self):
         # HACK: fix this to get along with NetworkCase.setUp(), and use super()
         MachineCase.setUp(self)
+        m = self.machine
+
+        # Prepare tmp directory
+        m.execute("mkdir {0}".format(self.tmp_storage))
+        self.addCleanup(m.execute, "rm -rf {0}".format(self.tmp_storage))
+
+        # Keep pristine state of libvirt
+        orig = "/var/lib/libvirt"
+        backup = os.path.join(self.tmp_storage, "libvirt.backup")
+        m.execute("cp -a {0} {1}".format(orig, backup))
+        self.addCleanup(m.execute, "rm -rf {0} && cp -a {1} {0}".format(orig, backup))
+
+        # Cleanup domains definitions
+        etc_orig = "/etc/libvirt"
+        etc_backup = os.path.join(self.tmp_storage, "etc_libvirt.backup")
+        m.execute("cp -a {0} {1}".format(etc_orig, etc_backup))
+        self.addCleanup(m.execute, "rm -rf {0} && cp -a {1} {0}".format(etc_orig, etc_backup))
+
+        # Cleanup pools
+        self.addCleanup(m.execute, "rm -rf /var/run/libvirt/storage/*")
+
+        # Cleanup networks
+        self.addCleanup(m.execute, "rm -rf /var/run/libvirt/network/test_network*")
+
         self.startLibvirt()
+        self.addCleanup(m.execute, "systemctl stop libvirtd")
+
+        # Stop all networks
+        self.addCleanup(m.execute, "for n in $(virsh net-list --all --name); do virsh net-destroy $n || true; done")
```

Comment:
That destroys the default network as well. But I suppose restoring /var/lib and /etc and restarting libvirt brings that back?

Reply:
Yeah, we do `net-start default` so it is fine.